uplink/src/base/serializer.rs

```diff
@@ -217,7 +316,7 @@ impl Serializer {
 
         let publish = match read(storage.reader(), max_packet_size) {
             Ok(Packet::Publish(publish)) => publish,
-            Ok(packet) => return Err(Error::UnexpectedPacket(packet)),
+            Ok(packet) => unreachable!("Unexpected packet: {:?}", packet),
```

Comment:
This branch implies storage has been corrupted, panic to stop `Serializer` completely